app/src/main/java/org/privacymatters/safespace/lib/fileManager/Operations.kt

```diff
@@ -70,53 +75,82 @@ class Operations(private val application: Application) {
 
     }
 
-    fun importFile(uri: Uri, internalPath: String): Int {
+    private suspend fun copyFileWithProgressNotifications(
+        sourceFileName: String,
+        sourceFileStream: InputStream,
+        targetFileStream: FileOutputStream,
+        type: FileTransferNotification.NotificationType,
+        context: Context
+    ) {
+        val uniqueNotificationId = sourceFileName.hashCode()
+        val fileTransferNotification = FileTransferNotification(context, uniqueNotificationId)
+        FileHelper.copyFileWithProgress(sourceFileStream, targetFileStream)
+            .catch { exception ->
+                exception.printStackTrace()
+                fileTransferNotification.showFailureNotification(sourceFileName, exception)
+            }
+            .collect { progress ->
+                fileTransferNotification.showProgressNotification(
+                    fileName = sourceFileName,
+                    progress = progress,
+                    type = type
+                )
+            }
+        fileTransferNotification.showSuccessNotification(
```

Comment:
While fiddling with it today, I realized that error notifications won't be correctly displayed because when an exception occurs during file transfer within a flow, it triggers a failure notification. However, simultaneously, the flow closes immediately, and the `FTN.showSuccessNotification()` gets called, replacing the error notification even before the user has a chance to see it. This results in showing the success notification in both success and failure scenarios, which isn't the desired behavior. Please wait for some time as I'll push a fix for this issue very soon.

Reply:
Sure take your time :)